File: examples/javascript/replace.mjs
#!/usr/bin/env node
import { EditableDocument } from 'office-oxide';

if (process.argv.length !== 4) {
  console.error('usage: replace.mjs <template> <output>');
  process.exit(1);
}

const ed = EditableDocument.open(process.argv[2]);
try {
  let n = ed.replaceText('{{NAME}}', 'Alice');
  n += ed.replaceText('{{DATE}}', '2026-04-18');
  console.log('replacements:', n);
  ed.save(process.argv[3]);
} finally {
  ed.close();
}

File: examples/python/replace.py
"""Fill a DOCX/PPTX template by replacing placeholders."""
import sys
from office_oxide import EditableDocument


def main(src: str, dst: str) -> None:
    with EditableDocument.open(src) as ed:
        n = ed.replace_text("{{NAME}}", "Alice")
        n += ed.replace_text("{{DATE}}", "2026-04-18")
        print(f"replacements: {n}")
        ed.save(dst)


if __name__ == "__main__":
    if len(sys.argv) != 3:
        sys.exit("usage: replace.py <template> <output>")
    main(sys.argv[1], sys.argv[2])
